python/ray/_private/runtime_env/packaging.py

```diff
@@ -281,15 +294,19 @@ def get_uri_for_directory(directory: str,
 
     hash_val = _hash_directory(directory, directory,
                                _get_excludes(directory, excludes))
+    return get_pkg_uri(
+        protocol=Protocol.GCS.value, pkg_name=RAY_PKG_PREFIX + hash_val.hex())
+
 
+def get_pkg_uri(protocol, pkg_name):
     return "{protocol}://{pkg_name}.zip".format(
-        protocol=Protocol.GCS.value, pkg_name=RAY_PKG_PREFIX + hash_val.hex())
+        protocol=protocol, pkg_name=pkg_name)
 
 
-def upload_package_to_gcs(pkg_uri: str, pkg_bytes: bytes):
-    protocol, pkg_name = parse_uri(pkg_uri)
+def upload_package_to_gcs(uri: str, pkg_bytes: bytes):
+    plugin, protocol, pkg_name = parse_uri(uri)
```

Comment:
Looks like Jobs is also using `upload_package_to_gcs`.  We could make jobs use `jobs|` as a prefix for its URIs, or think of some other way to reconcile this.